test/Microsoft.Azure.Devices.Edge.Test.Common/Windows/EdgeDaemon.cs

```diff
@@ -61,52 +58,78 @@ public EdgeDaemon(string scriptDir)
-            {
-                "$ProgressPreference='SilentlyContinue'",
-                $". {this.scriptDir}\\IotEdgeSecurityDaemon.ps1",
-                "Uninstall-IoTEdge -Force"
-            };
-            return Profiler.Run(
-                async () =>
+            return proxy.ForEachAsync(
+                p =>
                 {
-                    string[] output =
-                        await Process.RunAsync("powershell", string.Join(";", commands), token);
-                    Log.Verbose(string.Join("\n", output));
-                },
-                "Uninstalled edge daemon");
+                    return Profiler.Run(
+                        async () =>
+                        {
+                            await this.InternalStopAsync(token);
+
+                            var yaml = new DaemonConfiguration();
+                            yaml.AddHttpsProxy(p);
+                            yaml.Update();
+
+                            await this.InternalStartAsync(token);
+                        },
+                        "Configured edge daemon with proxy '{ProxyUri}'",
+                        p);
+                });
         }
 
         public Task StartAsync(CancellationToken token)
         {
-            var sc = new ServiceController("iotedge");
             return Profiler.Run(
-                async () =>
-                {
-                    if (sc.Status != ServiceControllerStatus.Running)
-                    {
-                        sc.Start();
-                        await this.WaitForStatusAsync(sc, ServiceControllerStatus.Running, token);
-                    }
-                },
+                () => this.InternalStartAsync(token),
                 "Started edge daemon");
         }
 
+        async Task InternalStartAsync(CancellationToken token)
+        {
+            var sc = new ServiceController("iotedge");
+            if (sc.Status != ServiceControllerStatus.Running)
+            {
+                sc.Start();
+                await this.WaitForStatusAsync(sc, ServiceControllerStatus.Running, token);
+            }
+        }
+
         public Task StopAsync(CancellationToken token)
+        {
+            return Profiler.Run(
+                () => this.InternalStopAsync(token),
+                "Stopped edge daemon");
+        }
+
+        async Task InternalStopAsync(CancellationToken token)
         {
             var sc = new ServiceController("iotedge");
+            if (sc.Status != ServiceControllerStatus.Stopped)
+            {
+                sc.Stop();
+                await this.WaitForStatusAsync(sc, ServiceControllerStatus.Stopped, token);
+            }
+        }
+
+        public Task UninstallAsync(CancellationToken token)
+        {
+            var commands = new[]
+            {
+                "$ProgressPreference='SilentlyContinue'",
+                $". {this.scriptDir}\\IotEdgeSecurityDaemon.ps1",
+                "Uninstall-IoTEdge -Force"
```

Comment:
uninstall-iotedge will uninstall iotedge moby, right? should we support an option to keep docker images in order to optimize e2e test run time? any idea?

Reply:
I think we should do some work like this, as an optimization to the builds, but not in this PR.